src/app/directives/datatable/datatable.directive.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  AfterContentInit,
+  Directive,
+  ElementRef,
+  Input,
+  OnInit
+} from "@angular/core";
+import { DatatableComponent } from "@swimlane/ngx-datatable";
+
+@Directive({
+  selector: "[appDatatable]"
+})
+export class DatatableDirective implements OnInit, AfterContentInit {
+  @Input() datatable: DatatableComponent;
```

Comment:
I believe you can use `@Host` here rather than `@Input` to get angular to inject whatever host component the directive is attached to